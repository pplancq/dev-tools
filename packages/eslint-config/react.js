module.exports = {
  extends: ['./rules/react.js', './rules/prettier.js'].map(require.resolve),
};
