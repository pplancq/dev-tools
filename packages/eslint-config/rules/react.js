module.exports = {
  extends: [
    'airbnb',
    'airbnb/hooks',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    './base.js',
    './typescript.js',
  ],
  rules: {
    // eslint-plugin-import https://github.com/import-js/eslint-plugin-import
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
      },
    ],

    // eslint-plugin-react https://github.com/jsx-eslint/eslint-plugin-react
    'react/no-multi-comp': [
      'error',
      {
        ignoreStateless: false,
      },
    ],
    'react/jsx-props-no-spreading': 'off',
    'react/no-unescaped-entities': 'off',
    'react/jsx-filename-extension': [
      'error',
      {
        extensions: ['.jsx'],
      },
    ],
    'react/function-component-definition': [
      'error',
      {
        namedComponents: ['function-declaration', 'arrow-function', 'function-expression'],
        unnamedComponents: ['arrow-function', 'function-expression'],
      },
    ],

    // eslint-plugin-react-hooks https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hook
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    react: {
      version: 'detect',
    },
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx'],
      },
    },
  },
  overrides: [
    {
      files: ['**/*.ts?(x)'],
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
      },
      rules: {
        // eslint-plugin-import https://github.com/import-js/eslint-plugin-import
        'import/extensions': [
          'error',
          'ignorePackages',
          {
            js: 'never',
            jsx: 'never',
            ts: 'never',
            tsx: 'never',
          },
        ],

        // eslint-plugin-react https://github.com/jsx-eslint/eslint-plugin-react
        'react/require-default-props': 'off',
        'react/jsx-filename-extension': [
          'error',
          {
            extensions: ['.jsx', '.tsx'],
          },
        ],
      },
    },
    {
      extends: ['plugin:jest-dom/recommended'],
      files: ['*.{test,spec,steps}.{js,jsx,ts,tsx}'],
      plugins: ['testing-library'],
      rules: {
        // eslint-plugin-testing-library https://github.com/testing-library/eslint-plugin-testing-library
        'testing-library/await-async-queries': 'error',
        'testing-library/await-async-utils': 'error',
        'testing-library/no-await-sync-queries': 'error',
        'testing-library/no-container': 'error',
        'testing-library/no-debugging-utils': 'error',
        'testing-library/no-dom-import': ['error', 'react'],
        'testing-library/no-node-access': 'error',
        'testing-library/no-promise-in-fire-event': 'error',
        'testing-library/no-render-in-lifecycle': 'error',
        'testing-library/no-unnecessary-act': 'error',
        'testing-library/no-wait-for-multiple-assertions': 'error',
        'testing-library/no-wait-for-side-effects': 'error',
        'testing-library/no-wait-for-snapshot': 'error',
        'testing-library/prefer-find-by': 'error',
        'testing-library/prefer-presence-queries': 'error',
        'testing-library/prefer-query-by-disappearance': 'error',
        'testing-library/prefer-screen-queries': 'error',
        'testing-library/render-result-naming-convention': 'error',
      },
    },
  ],
};
