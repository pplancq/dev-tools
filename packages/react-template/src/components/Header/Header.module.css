.container {
  display: flex;
  width: auto;
  padding: 8px;
  flex-direction: row;
  align-items: center;
}
