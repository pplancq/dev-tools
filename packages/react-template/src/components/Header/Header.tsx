import logo from '@Front/assets/images/logo.png';
import { Logo } from '@Front/ui/atoms/Logo';
import { Navigation } from '@Front/ui/molecules/Navigation';
import { Link } from 'react-router-dom';

import classes from './Header.module.css';

export const Header = () => {
  return (
    <div className={classes.container}>
      <Link to="/">
        <Logo size="50px" src={logo} alt="logo" />
      </Link>
      <Navigation links={[{ label: 'Demo', url: '/' }]} />
    </div>
  );
};
