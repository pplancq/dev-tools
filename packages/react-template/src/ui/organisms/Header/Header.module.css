header {
  display: flex;
  flex-direction: row;
  flex-grow: 0;
  flex-shrink: 0;
  align-items: center;
  justify-content: space-between;
  background-color: #00008f;
}

@media screen and (width <= 900px) {
  header {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
