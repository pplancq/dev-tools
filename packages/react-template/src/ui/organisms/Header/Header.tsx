import { ReactNode } from 'react';
import classes from './Header.module.css';

type HeaderProps = {
  leftComponents?: ReactNode;
  rightComponents?: ReactNode;
};

export const Header = ({ leftComponents, rightComponents }: HeaderProps) => {
  return (
    <header className={classes.header}>
      {leftComponents}
      {rightComponents}
    </header>
  );
};
