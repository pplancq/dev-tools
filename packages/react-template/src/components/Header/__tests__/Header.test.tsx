import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Header } from '../Header';

describe('Header Component', () => {
  it('should render the Header component correctly', () => {
    const { container } = render(
      <Router future={{ v7_startTransition: true, v7_relativeSplatPath: true }}>
        <Header />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });
});
