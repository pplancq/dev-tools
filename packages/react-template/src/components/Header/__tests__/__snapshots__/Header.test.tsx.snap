// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > should render the Header component correctly 1`] = `
<div>
  <div
    class="_container_63c2ea"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        class="_logoStyle_a0661f"
        src="/src/assets/images/logo.png"
        style="width: 50px;"
      />
    </a>
    <nav>
      <ul
        class="_navList_75d6d6"
      >
        <li>
          <a
            aria-current="page"
            class="_a_75d6d6 active"
            href="/"
          >
            Demo
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
